Upgrade ember, ember-cli and ember-data

jpadilla opened the pull request.

gcollazo commented:
When you do a search that returns no results and the {{else}} inside the {{each}} is invoked everything works fine. But if you reset the search and ember tries to re-render the list we get an exception and the list rendering stops.

> Uncaught TypeError: Cannot read property 'insertBefore' of null

**Steps to reproduce:**

- Search for something that returns no results like, "zzzzz".
- Reset the search so that ember tries to re-render the list.
- Check your console.

Also tried to move to the <em-pkg pkg={{pkg}}> syntax, but we need the `tagName` to be a `<tr>` and that doesn't work either. The `tagName` property on the `component.js` has no effect on the DOM that is produced.

I think we are suffering from issue https://github.com/emberjs/ember.js/issues/11164.

rwjblue commented:
It is almost definitely https://github.com/emberjs/ember.js/issues/11140, not https://github.com/emberjs/ember.js/issues/11164 (unless you are using `itemViewClass` / `itemController` or another hash parameter for `{{each}}` that causes us to fallback to `LegacyEachView`).

gcollazo commented:
Yes you are right, we are not using `itemViewClass` / `itemController`.

jpadilla commented:
emberjs/ember.js#11140 is now fixed in canary and fixed our original issue here.

Went ahead and updated ember to canary, updated ember-cli to 0.2.7, and ember-data to 1.0.0-beta.18.

I ran into issues with ember-data >= 1.0.0-beta.19 and `computedFilterByQuery` where the results returned in `filteredPackages` was an array of `InternalModel` instead of `DS.Model`.
